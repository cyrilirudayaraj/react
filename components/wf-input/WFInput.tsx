import React from 'react';
import { Label, Input, FormError } from '@athena/forge';
import { Field, ErrorMessage } from 'formik';
import { Error } from '../error/Error';

const WFInput = (props: any): JSX.Element => {
  return (
    <div className="input-wrapper">
      <Label text={props.label} className="fe_u_margin--top-medium" />
      <Field
        as={Input}
        id={props.name}
        name={props.name}
        required={props.required === undefined ? true : props.required}
        disabled={props.disabled}
        value={props.value}
        className="fe_u_margin--top-xsmall fe_u_margin--bottom-medium"
      />
      {props.showError === undefined ? (
        <ErrorMessage name={props.name} component={Error} />
      ) : props.showError ? (
        <div>
          <FormError>{props.errorMessage}</FormError>
        </div>
      ) : null}
    </div>
  );
};

export default WFInput;
