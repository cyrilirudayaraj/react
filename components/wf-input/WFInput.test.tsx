import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WFInput from './WFInput';

configure({ adapter: new Adapter() });
let wrapper: any;

describe('snapshot test <WFInput>', () => {
  it('should render as expected', () => {
    wrapper = shallow(<WFInput label="Title" name="title" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should render required property', () => {
    wrapper = shallow(
      <WFInput label="Title" name="title" required={false} showError={true} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should render with show error as false', () => {
    wrapper = shallow(
      <WFInput label="Title" name="title" required={false} showError={false} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
