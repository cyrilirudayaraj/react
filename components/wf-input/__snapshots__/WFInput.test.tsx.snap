// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFInput> should render as expected 1`] = `
"<div className=\\"input-wrapper\\">
  <Label text=\\"Title\\" className=\\"fe_u_margin--top-medium\\" required={false} useLegend={false} />
  <Field as={{...}} id=\\"title\\" name=\\"title\\" required={true} disabled={[undefined]} value={[undefined]} className=\\"fe_u_margin--top-xsmall fe_u_margin--bottom-medium\\" />
  <FormikConnect(ErrorMessageImpl) name=\\"title\\" component={[Function: Error]} />
</div>"
`;

exports[`snapshot test <WFInput> should render required property 1`] = `
"<div className=\\"input-wrapper\\">
  <Label text=\\"Title\\" className=\\"fe_u_margin--top-medium\\" required={false} useLegend={false} />
  <Field as={{...}} id=\\"title\\" name=\\"title\\" required={false} disabled={[undefined]} value={[undefined]} className=\\"fe_u_margin--top-xsmall fe_u_margin--bottom-medium\\" />
  <div>
    <FormError alertType=\\"attention\\" />
  </div>
</div>"
`;

exports[`snapshot test <WFInput> should render with show error as false 1`] = `
"<div className=\\"input-wrapper\\">
  <Label text=\\"Title\\" className=\\"fe_u_margin--top-medium\\" required={false} useLegend={false} />
  <Field as={{...}} id=\\"title\\" name=\\"title\\" required={false} disabled={[undefined]} value={[undefined]} className=\\"fe_u_margin--top-xsmall fe_u_margin--bottom-medium\\" />
</div>"
`;
